### ISSUE TITLE
Python 3.11: ImportError: cannot import name 'getargspec' from 'inspect' 

### ISSUE THREAD

**Issue by ivankravets** [bottlepy/bottle#1387]
See related issue https://github.com/platformio/platformio-vscode-ide/issues/3252

**Comment by defnull** [comment]
Bottle (0.12 or dev) does not import `inspect.getargspec` in a python 3 environment. Which version of bottle are you running?

**Comment by ivankravets** [comment]
0.12.*, see https://github.com/platformio/platformio-core/blob/develop/setup.py#L30

I've just checked the source code of `0.12.21` and I see 

```python
from traceback import format_exc, print_exc
from inspect import getargspec
from unicodedata import normalize
```

**Comment by ivankravets** [comment]
See https://github.com/bottlepy/bottle/blob/0.12.21/bottle.py#L44

**Comment by defnull** [comment]
Ahh, this was already fixed in `release-0.12` but not yet released to PyPi. Sorry, I'll prepare a release.

**Comment by defnull** [comment]
Fixed in 0.12.23
